### PR TITLE
Add Release Workflow and Build Script for Automated Releases

### PR DESCRIPTION
**Pull request by maxlambrecht**
**Pull request check list**

- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

**Description of change**

This pull request introduces two new files that automate the release process for the Galadriel project. These additions will streamline the creation and publishing of new releases, ensuring that they are consistent, reliable, and require minimal manual intervention.

- GitHub Release Workflow `release_build.yaml`: This GitHub Actions workflow is triggered whenever a new version tag (formatted as `vX.Y.Z`) is pushed to the repository. The workflow sets up the Go environment, checks out the code, downloads dependencies, runs tests, and builds the project's artifacts using the provided build script. It then creates a new GitHub release and uploads the built artifacts and their SHA-256 checksums to it. The workflow is designed to run on an Ubuntu-latest environment.

- Build Artifacts Script `build_artifacts.sh`: This bash script is responsible for building the Galadriel project's artifacts. It supports building for two architectures: `amd64` and `arm64`. For each architecture, it builds the `galadriel-server` and `galadriel-harvester` binaries, packages them along with the `LICENSE` and configuration files into a tarball, and generates a SHA-256 checksum for the tarball. The script is designed to halt on the first error encountered to prevent the propagation of build errors.

**Which issue this pull requests fixes**

### CODE REVIEW

**Review by Victorblsilveira — COMMENTED** [review]
Just some comments and possible improvements.

File: .github/workflows/release_build.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Release Build
```
[the remark this answers is not in the document]
Comment (maxlambrecht) [reply .github/workflows/release_build.yaml:1]: I refactored the flow in multiple jobs to set specific permissions.

File: .github/workflows/scripts/build_artifacts.sh
[the remark this answers is not in the document]
Comment (maxlambrecht) [reply .github/workflows/scripts/build_artifacts.sh:1]: Thank you for your suggestion. While I agree that the SLSA PR could potentially provide valuable improvements to our workflow, I believe it's beyond the scope of this particular PR. Our current focus is on implementing the release workflow and ensuring it functions as expected. Once the SLSA PR is completed and merged, we can certainly revisit our workflow and consider incorporating any relevant enhancements.

[the remark this answers is not in the document]
Comment (maxlambrecht) [reply .github/workflows/scripts/build_artifacts.sh:1]: What I did following the SLSA PR is to change my initial script to use the Makefile to build the binaries.

**Pull request merged**